### PR TITLE
Feature/heroku script

### PR DESCRIPTION
**Pull request by muhtarudinsiregar**
Hi @magnobiet can you take a look my code? for this issue #3

### CODE REVIEW

**Comment by magnobiet** [comment]
👏  Great job @muhtarudinsiregar, thank you so much.

File: app.json [hunk 1]
```diff
@@ -1,33 +1,36 @@
...
-    "APP_DEBUG": "true",
-    "HEROKU_APP_NAME": {
-      "required": true
+    "name": "laravel-bootstrap",
+    "scripts": {
+        "postdeploy": "php artisan key:generate && php artisan db:seed && php artisan app:release"
     },
-    "HEROKU_PARENT_APP_NAME": {
-      "required": true
+    "env": {
```
Comment (magnobiet) [inline app.json]: Leave only `APP_DEBUG` and `APP_ENV` as a required field.

**Comment by magnobiet** [comment]
Build by button "Deploy to Heroku" is failed.

1. Your 'composer.lock' is out of date!
2.  In ProviderRepository.php line 208: `Class 'Barryvdh\Debugbar\ServiceProvider' not found`

Can you see that?

### Log file

```log
-----> Node.js app detected



-----> Creating runtime environment



			 NPM_CONFIG_LOGLEVEL=error

			 NODE_ENV=production

			 NODE_MODULES_CACHE=true

			 NODE_VERBOSE=false



-----> Installing binaries

			 engines.node (package.json):  unspecified

			 engines.npm (package.json):   unspecified (use default)



			 Resolving node version 8.x...

			 Downloading and installing node 8.12.0...

			 Using default npm version: 6.4.1



-----> Building dependencies

			 Installing node modules (package.json + package-lock)



			 > node-sass@4.9.0 install /tmp/build_1c0006ff9901401369c55a24a0758afc/node_modules/node-sass

			 > node scripts/install.js



			 Downloading binary from https://github.com/sass/node-sass/releases/download/v4.9.0/linux-x64-57_binding.node

			 Download complete

			 Binary saved to /tmp/build_1c0006ff9901401369c55a24a0758afc/node_modules/node-sass/vendor/linux-x64-57/binding.node

			 Caching binary to /tmp/npmcache.HGwF5/node-sass/4.9.0/linux-x64-57_binding.node



			 > uglifyjs-webpack-plugin@0.4.6 postinstall /tmp/build_1c0006ff9901401369c55a24a0758afc/node_modules/webpack/node_modules/uglifyjs-webpack-plugin

			 > node lib/post_install.js





			 > node-sass@4.9.0 postinstall /tmp/build_1c0006ff9901401369c55a24a0758afc/node_modules/node-sass

			 > node scripts/build.js



			 Binary found at /tmp/build_1c0006ff9901401369c55a24a0758afc/node_modules/node-sass/vendor/linux-x64-57/binding.node

			 Testing binary

			 Binary is fine



			 > gifsicle@3.0.4 postinstall /tmp/build_1c0006ff9901401369c55a24a0758afc/node_modules/gifsicle

			 > node lib/install.js



				 ✔ gifsicle pre-build test passed successfully



			 > mozjpeg@5.0.0 postinstall /tmp/build_1c0006ff9901401369c55a24a0758afc/node_modules/mozjpeg

			 > node lib/install.js



				 ✔ mozjpeg pre-build test passed successfully



			 > optipng-bin@3.1.4 postinstall /tmp/build_1c0006ff9901401369c55a24a0758afc/node_modules/optipng-bin

			 > node lib/install.js



				 ✔ optipng pre-build test passed successfully



			 > pngquant-bin@4.0.0 postinstall /tmp/build_1c0006ff9901401369c55a24a0758afc/node_modules/pngquant-bin

			 > node lib/install.js



				 ⚠ The `/tmp/build_1c0006ff9901401369c55a24a0758afc/node_modules/pngquant-bin/vendor/pngquant` binary doesn't seem to work correctly

				 ⚠ pngquant pre-build test failed

				 ℹ compiling from source

				 ✔ pngquant pre-build test passed successfully

				 ✔ pngquant built successfully



			 > undefined install /tmp/build_1c0006ff9901401369c55a24a0758afc

			 > napa



			 info download https://github.com/fronteed/iCheck/archive/1.0.2.tar.gz into icheck



			 > undefined postinstall /tmp/build_1c0006ff9901401369c55a24a0758afc

			 > npm run prod





			 > @ prod /tmp/build_1c0006ff9901401369c55a24a0758afc

			 > npm run production





			 > @ production /tmp/build_1c0006ff9901401369c55a24a0758afc

			 > cross-env NODE_ENV=production node_modules/webpack/bin/webpack.js --no-progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js



				DONE  Compiled successfully in 15367ms23:40:21



																																																 Asset     Size  Chunks             Chunk Names

				 fonts/vendor/bootstrap/dist/glyphicons-halflings-regular.eot?f4769f9bdb7466be65088239c12046d1  20.1 kB          [emitted]

			 fonts/vendor/bootstrap/dist/glyphicons-halflings-regular.woff2?448c34a56d699c29117adc64c43affeb    18 kB          [emitted]

				fonts/vendor/bootstrap/dist/glyphicons-halflings-regular.woff?fa2772327f55d8198301fdb8bcfc8158  23.4 kB          [emitted]

				 fonts/vendor/bootstrap/dist/glyphicons-halflings-regular.ttf?e18bbf611f2a2e43afc071aa2f4e1512  45.4 kB          [emitted]

				 fonts/vendor/bootstrap/dist/glyphicons-halflings-regular.svg?89889688147bd7575d6327160d64e760   109 kB          [emitted]

														images/vendor/admin-lte/dist/boxed-bg.jpg?5bbbd344130f0926b70aa3499638a387  27.2 kB          [emitted]

													 images/vendor/icheck/skins/square/blue.png?3bae04c2e0b224696c461e0a7b5d1e99  2.01 kB          [emitted]

												images/vendor/icheck/skins/square/blue@2x.png?143101c35ce1275fd7b143a9ef68f8c1  4.04 kB          [emitted]

																																											 /scripts/app.js  81.5 kB       0  [emitted]  /scripts/app

																																											 /styles/app.css   245 kB    0, 0  [emitted]  /scripts/app, /scripts/app

			 added 1563 packages from 795 contributors and audited 21237 packages in 64.043s

			 found 45 vulnerabilities (21 low, 21 moderate, 2 high, 1 critical)

				 run `npm audit fix` to fix them, or `npm audit` for details



-----> Caching build

			 - node_modules



-----> Pruning devDependencies

			 removed 1390 packages and audited 1810 packages in 16.951s

			 found 1 moderate severity vulnerability

				 run `npm audit fix` to fix them, or `npm audit` for details



-----> Build succeeded!

 !     This app may not specify any way to start a node process

			 https://devcenter.heroku.com/articles/nodejs-support#default-web-process-type

-----> PHP app detected

-----> Bootstrapping...

 !     WARNING: Your 'composer.lock' is out of date!

 !

 !     The 'composer.lock' file in your project is not up to date with

 !     the main 'composer.json' file. This may result in installation

 !     of incorrect packages or package versions.

 !

 !     The lock file is required in order to guarantee reliable and

 !     reproducible installation of dependencies across systems and

 !     deploys. It must always be kept in sync with 'composer.json'.

 !

 !     Whenever you change 'composer.json', ensure that you perform

 !     the following steps locally on your computer:

 !     1) run 'composer update'

 !     2) add all changes using 'git add composer.json composer.lock'

 !     3) commit using 'git commit'

 !

 !     Ensure that you updated the lock file correctly, and that you

 !     ran 'git add' on both files, before deploying again.

 !

 !     Please remember to always keep your 'composer.lock' updated in

 !     lockstep with 'composer.json' to avoid common problems related

 !     to dependencies during collaboration and deployment.

 !

 !     Please refer to the Composer documentation for further details:

 !     https://getcomposer.org/doc/

 !     https://getcomposer.org/doc/01-basic-usage.md

-----> Installing platform packages...

			 - php (7.2.10)

			 - ext-mbstring (bundled with php)

			 - nginx (1.8.1)

			 - apache (2.4.34)

-----> Installing dependencies...

			 Composer version 1.7.2 2018-08-16 16:57:12

			 Loading composer repositories with package information

			 Installing dependencies from lock file

			 Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. Run update to update them.

			 Package operations: 63 installs, 0 updates, 0 removals

				 - Installing psr/log (1.0.2): Downloading (100%)

				 - Installing vlucas/phpdotenv (v2.4.0): Downloading (100%)

				 - Installing symfony/css-selector (v4.0.9): Downloading (100%)

				 - Installing tijsverkoyen/css-to-inline-styles (2.2.1): Downloading (100%)

				 - Installing symfony/polyfill-php72 (v1.8.0): Downloading (100%)

				 - Installing symfony/polyfill-mbstring (v1.8.0): Downloading (100%)

				 - Installing symfony/var-dumper (v4.0.9): Downloading (100%)

				 - Installing symfony/routing (v4.0.9): Downloading (100%)

				 - Installing symfony/process (v4.0.9): Downloading (100%)

				 - Installing symfony/http-foundation (v4.0.9): Downloading (100%)

				 - Installing symfony/event-dispatcher (v4.0.9): Downloading (100%)

				 - Installing symfony/debug (v4.0.9): Downloading (100%)

				 - Installing symfony/http-kernel (v4.0.9): Downloading (100%)

				 - Installing symfony/finder (v4.0.9): Downloading (100%)

				 - Installing symfony/console (v4.0.9): Downloading (100%)

				 - Installing doctrine/lexer (v1.0.1): Downloading (100%)

				 - Installing egulias/email-validator (2.1.4): Downloading (100%)

				 - Installing swiftmailer/swiftmailer (v6.0.2): Downloading (100%)

				 - Installing paragonie/random_compat (v2.0.12): Downloading (100%)

				 - Installing ramsey/uuid (3.7.3): Downloading (100%)

				 - Installing psr/simple-cache (1.0.1): Downloading (100%)

				 - Installing psr/container (1.0.0): Downloading (100%)

				 - Installing symfony/translation (v4.0.9): Downloading (100%)

				 - Installing nesbot/carbon (1.25.0): Downloading (100%)

				 - Installing monolog/monolog (1.23.0): Downloading (100%)

				 - Installing league/flysystem (1.0.44): Downloading (100%)

				 - Installing erusev/parsedown (1.7.1): Downloading (100%)

				 - Installing dragonmantank/cron-expression (v2.1.0): Downloading (100%)

				 - Installing doctrine/inflector (v1.3.0): Downloading (100%)

				 - Installing laravel/framework (v5.6.19): Downloading (100%)

				 - Installing arcanedev/support (4.3.0): Downloading (100%)

				 - Installing arcanedev/log-viewer (4.5.1): Downloading (100%)

				 - Installing fideloper/proxy (4.0.0): Downloading (100%)

				 - Installing guzzlehttp/promises (v1.3.1): Downloading (100%)

				 - Installing psr/http-message (1.0.1): Downloading (100%)

				 - Installing guzzlehttp/psr7 (1.4.2): Downloading (100%)

				 - Installing jakub-onderka/php-console-color (0.1): Downloading (100%)

				 - Installing jeroennoten/laravel-adminlte (v1.23.1): Downloading (100%)

				 - Installing cloudinary/cloudinary_php (1.9.0): Downloading (100%)

				 - Installing jrm2k6/cloudder (0.5.1): Downloading (100%)

				 - Installing kkomelin/laravel-translatable-string-exporter (0.4.0): Downloading (100%)

				 - Installing zendframework/zend-diactoros (1.7.1): Downloading (100%)

				 - Installing symfony/psr-http-message-bridge (v1.0.2): Downloading (100%)

				 - Installing phpseclib/phpseclib (2.0.11): Downloading (100%)

				 - Installing league/event (2.1.2): Downloading (100%)

				 - Installing lcobucci/jwt (3.2.2): Downloading (100%)

				 - Installing defuse/php-encryption (v2.2.0): Downloading (100%)

				 - Installing league/oauth2-server (7.1.0): Downloading (100%)

				 - Installing guzzlehttp/guzzle (6.3.3): Downloading (100%)

				 - Installing firebase/php-jwt (v5.0.0): Downloading (100%)

				 - Installing laravel/passport (v6.0.0): Downloading (100%)

				 - Installing nikic/php-parser (v4.0.1): Downloading (100%)

				 - Installing jakub-onderka/php-console-highlighter (v0.3.2): Downloading (100%)

				 - Installing dnoegel/php-xdg-base-dir (0.1): Downloading (100%)

				 - Installing psy/psysh (v0.9.3): Downloading (100%)

				 - Installing laravel/tinker (v1.0.6): Downloading (100%)

				 - Installing league/fractal (0.17.0): Downloading (100%)

				 - Installing owen-it/laravel-auditing (v6.1.1): Downloading (100%)

				 - Installing prettus/laravel-validation (1.1.5): Downloading (100%)

				 - Installing prettus/l5-repository (2.6.32): Downloading (100%)

				 - Installing robclancy/presenter (1.3.2): Downloading (100%)

				 - Installing rollbar/rollbar (v1.4.3): Downloading (100%)

				 - Installing rollbar/rollbar-laravel (v2.3.0): Downloading (100%)

			 Generating optimized autoload files

			 > Illuminate\Foundation\ComposerScripts::postAutoloadDump

			 > @php artisan package:discover



			 In ProviderRepository.php line 208:



				 Class 'Barryvdh\Debugbar\ServiceProvider' not found





			 Script @php artisan package:discover handling the post-autoload-dump event returned with error code 1

 !     WARNING: There was a class not found error in your code

 !     ERROR: Dependency installation failed!

 !

 !     The 'composer install' process failed with an error. The cause

 !     may be the download or installation of packages, or a pre- or

 !     post-install hook (e.g. a 'post-install-cmd' item in 'scripts')

 !     in your 'composer.json'.

 !

 !     Typical error cases are out-of-date or missing parts of code,

 !     timeouts when making external connections, or memory limits.

 !

 !     Check the above error output closely to determine the cause of

 !     the problem, ensure the code you're pushing is functioning

 !     properly, and that all local changes are committed correctly.

 !

 !     For more information on builds for PHP on Heroku, refer to

 !     https://devcenter.heroku.com/articles/php-support

 !

 !     REMINDER: the following warnings were emitted during the build;

 !     check the details above, as they may be related to this error:

 !     - Your 'composer.lock' is out of date!

 !     - There was a class not found error in your code

 !     Push rejected, failed to compile PHP app.

 !     Push failed
```

File: app.json (hunk 1 above)
Comment (muhtarudinsiregar) [reply app.json]: so HEROKU_PARENT_APP_NAME and other, can i remove it? or you have example?

Comment (magnobiet) [reply app.json:6]: Leave it, only change to `"required": false`.

https://devcenter.heroku.com/articles/app-json-schema#env

**Review by magnobiet — APPROVED** [review]

**Comment by magnobiet** [comment]
Thank you @muhtarudinsiregar.

**Pull request merged**